src/lib/recursive_match.js

```diff
@@ -4,33 +4,40 @@
 function find_recursive(opener, closer, text) {
   var out = [];
   var last = [];
-  var chars = text.split('');
+  const chars = text.split('');
   var open = 0;
   for (var i = 0; i < chars.length; i++) {
+    const c = text[i];
     //increment open tag
-    if (chars[i] === opener) {
+    if (c === opener) {
       open += 1;
     }
     //decrement close tag
-    if (chars[i] === closer) {
+    else if (c === closer) {
       open -= 1;
       if (open < 0) {
         open = 0;
       }
+    } else if (last.length == 0) {
+      // If we're not inside of a pair of delimiters, we can discard the current letter.
```

Comment:
I think the only consumer of this function is https://github.com/spencermountain/wtf_wikipedia/blob/622dd48eb8bc1611447e87bd4321fae19dc567a1/src/03-paragraph/index.js#L26-L28, which will ignore the single character entries in the array, so we can just omit them and avoid paying the intermediate array operations.

I think. How confident should I be in the test suite to catch me breaking this function? :)